manual/manual/tutorials/coreexamples.etex

```diff
@@ -550,10 +550,16 @@ let assoc_may_map f x l =
   | y -> f y;;
 \end{caml_example}
 Note that this construction is only useful if the exception is raised
-between "match"\ldots"with". Moreover, exceptions can only
-appear at the toplevel of such a pattern match.
-For instance, exception cases cannot currently be combined with an
-or-pattern: "exception A | exception B ->" \ldots .
+between "match"\ldots"with". Multiple exception patterns can be combined
+together,
+\begin{caml_example}{toplevel}
+let with_option f x =
+  match f x with
+  | exception Not_found | exception Exit -> None
```

Comment:
I think it would be better to have `exception A | exception B`, since catching `Exit` here is a bit odd.